apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/RetriableTransactionOptionsTests.cs

```diff
@@ -24,38 +24,44 @@ public class RetriableTransactionOptionsTests
         public void DefaultJitter()
         {
             var options = RetriableTransactionOptions.CreateDefault();
+
             var delay = TimeSpan.FromTicks(TimeSpan.TicksPerMillisecond);
+            var delayTicks = (int)delay.Ticks;
+            // The jitter on RetriableTransactionOptions should jitter between
+            // delay and delay * 2.
+            var expectedMinDelayTicks = delayTicks;
+            var expectedMaxDelayTicks = delayTicks * 2;
+            var possibleDelayValues = expectedMaxDelayTicks - expectedMinDelayTicks;
+            var expectedMedianDelayTicks = expectedMinDelayTicks + possibleDelayValues / 2;
 
-            var delays = Enumerable.Range(0, (int)TimeSpan.TicksPerMillisecond)
+            var delays = Enumerable.Range(0, possibleDelayValues)
```

Comment:
This is still using a value which is fundamentally "a number of ticks" as "the number of iterations to perform" - they don't seem related to me, or at least, I don't understand why they would be.

It feels like the number of iterations could be any reasonably large number (even just 1000) - all we're checking is that at least one value is in the lower half and at least one value is in the upper half. I may be missing something though.

Reply:
> This is still using a value which is fundamentally "a number of ticks" as "the number of iterations to perform" - they don't seem related to me, or at least, I don't understand why they would be.

But ticks is as granular as TimeSpan would go. So, when we jitter, the quantity of possible values we can ever get is always the number of ticks between the jitter min and max bounds.
In a perfect random distribution we would get each of the possible values if we had max - min iterations, and that's why, yes, this is fundamentally a number of ticks, but it represents the amount of different values so it's ok to use it as the number of iterations to be performed. What is true is that, if the random generator is good, this tests is really loose, and we could probably do with way less iterations, or with being way more strict on our checks.

I would leave it as is, but have no problem with putting just a number there, LMK.

Reply:
I'd definitely prefer to have just a number. Even with a perfect random number generator, you wouldn't necessarily get each value exactly once - just like if you roll a fair die 6 times, you wouldn't always expect to get 1, 2, 3, 4, 5, 6.
(To put it another way - if that *were* the case, the last sample *wouldn't* be random, because you'd be able to predict the exact value - that would be weird!)

It might be reasonable to take, say, 10,000 samples and assert that between 4,000 and 6,000 of them should be in the lower half of the range - that would be a reasonable test that it's probably genuinely linear.

Reply:
Yep, true, it would be a perfect distribution, but not random :).
Changing the test now.